pkg/kubelet/kuberuntime/kuberuntime_logs.go

```diff
@@ -196,6 +204,40 @@ func ReadLogs(path string, apiOpts *v1.PodLogOptions, stdout, stderr io.Writer)
 	}
 }
 
+// waitLogs wait for the next log write. If the container is not running
+func (m *kubeGenericRuntimeManager) waitLogs(id string, w *fsnotify.Watcher) error {
+	errRetry := 5
+	for {
+		select {
+		case e := <-w.Events:
+			switch e.Op {
+			case fsnotify.Write:
+				return nil
+			default:
+				glog.Errorf("Unexpected fsnotify event: %v, retrying...", e)
+			}
+		case err := <-w.Errors:
+			glog.Errorf("Fsnotify watch error: %v, %d error retries remaining", err, errRetry)
+			if errRetry == 0 {
+				return err
+			}
+			errRetry--
+		case <-time.After(stateCheckPeriod):
+			s, err := m.runtimeService.ContainerStatus(id)
+			if err != nil {
+				return err
+			}
+			// Only keep following container log when it is running. There may be a race condition
```

Comment:
If the container is not running, we can't tail it anyway. Do we need to care about that?